ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/cache/CachedWorldStorageManager.java

```diff
@@ -151,7 +151,8 @@ public Optional<BonsaiWorldState> getHeadWorldState(
               addCachedLayer(
                   blockHeader,
                   blockHeader.getStateRoot(),
-                  new BonsaiWorldState(archive, rootWorldStateStorage));
+                  new BonsaiWorldState(archive,
+                      new BonsaiSnapshotWorldStateKeyValueStorage(rootWorldStateStorage)));
```

Comment:
@matkt IMO we should not add the persisted state to the cache directly, since it is mutable.  testing now with this tweak to ensure it does not re-introduce any leaks

Reply:
no new snapshot leakage found during testing with this commit